GVFS/GVFS/CommandLine/IncludeVerb.cs

```diff
@@ -0,0 +1,182 @@
+        public string Remove { get; set; }
+
+        [Option(
+            'l',
+            "list",
+            Required = false,
+            Default = false,
+            HelpText = "List of folders to for being included in projection.")]
+        public bool List { get; set; }
+
+        protected override string VerbName => IncludeVerbName;
+
+        protected override void Execute(GVFSEnlistment enlistment)
+        {
+            using (JsonTracer tracer = new JsonTracer(GVFSConstants.GVFSEtwProviderName, "Include"))
+            {
+                try
+                {
+                    tracer.AddLogFileEventListener(
+                        GVFSEnlistment.GetNewGVFSLogFileName(enlistment.GVFSLogsRoot, GVFSConstants.LogFileTypes.Include),
+                        EventLevel.Informational,
+                        Keywords.Any);
+
+                    using (GVFSDatabase database = new GVFSDatabase(new PhysicalFileSystem(), enlistment.EnlistmentRoot, new SqliteDatabase()))
+                    {
+                        IncludedFolderTable includedFolderTable = new IncludedFolderTable(database);
+                        bool haveFoldersToAdd = !string.IsNullOrEmpty(this.Add);
+                        bool haveFoldersToRemove = !string.IsNullOrEmpty(this.Remove);
+
+                        if (this.List || (!haveFoldersToAdd && !haveFoldersToRemove))
+                        {
+                            List<string> directories = includedFolderTable.GetAll();
+                            if (directories.Count == 0)
+                            {
+                                this.Output.WriteLine("No folders in included list.");
+                            }
+                            else
+                            {
+                                foreach (string directory in directories)
+                                {
+                                    this.Output.WriteLine(directory);
+                                }
+                            }
+
+                            return;
+                        }
+
+                        // Make sure there is a clean git status before allowing inclusions to change
+                        this.CheckGitStatus(tracer, enlistment);
+
+                        if (haveFoldersToRemove)
+                        {
+                            foreach (string directoryPath in this.Remove.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries))
+                            {
+                                includedFolderTable.Remove(directoryPath);
+                            }
+                        }
+
+                        if (haveFoldersToAdd)
+                        {
+                            foreach (string directoryPath in this.Add.Split(new[] { ';' }, StringSplitOptions.RemoveEmptyEntries))
+                            {
+                                includedFolderTable.Add(directoryPath);
+                            }

```

Comment:
I'm not sure that really matters.  What would the user action be from those messages?  Also it would be more queries to the database because we would need to check each path before taking any action.  The only thing I can think of this being useful for is multiple scripts that might be adding the same paths to go clean up the scripts.  I'm going to hold off on this for now until we get some real users feedback.  Let me know if you want me to create an issue for this enhancement.